### ISSUE TITLE
Unable to process nested `${}` expressions 

### ISSUE THREAD

**Issue by edrd-f** [casid/jte#112]
Hi! First of all, thanks for this great project.

I'm using it to generate Kotlin code and faced some problems since `${}` in Kotlin also means expression interpolation and therefore nesting is common. Take the following code as example:

###### Data.kt
```kotlin
data class Data(val now: LocalDateTime, val datePattern: String)
```

###### Template.kte
```
@import java.time.format.DateTimeFormatter
@import sample.Data
@param data: Data

println("Today is ${data.now.format(DateTimeFormatter.ofPattern("${data.datePattern} HH:mm")}")
```

JTE generates this:

```kotlin
class JteIssueGenerated {
companion object {
	@JvmField val JTE_NAME = "Issue.kte"
	@JvmField val JTE_LINE_INFO = intArrayOf(0,0,1,3,3,3,3,5,5,5,6)
	@JvmStatic fun render(jteOutput:gg.jte.TemplateOutput, jteHtmlInterceptor:gg.jte.html.HtmlInterceptor?, data:Data) {
		jteOutput.writeContent("\nprintln(\"")
		jteOutput.writeUserContent(data.now.format(DateTimeFormatter.ofPattern("${data.datePattern)
		jteOutput.writeContent(" HH:mm\"))}\")\n")
	}
	@JvmStatic fun renderMap(jteOutput:gg.jte.TemplateOutput, jteHtmlInterceptor:gg.jte.html.HtmlInterceptor?, params:Map<String, Any?>) {
		val data = params["data"] as Data
		render(jteOutput, jteHtmlInterceptor, data);
	}
}
}
```

Workaround is to use variable instead of expression interpolation, so the previous example would look like this:

```
@import java.time.format.DateTimeFormatter
@import sample.Data
@param data: Data

!{val datePattern = data.datePattern}
println("Today is ${data.now.format(DateTimeFormatter.ofPattern("$datePattern HH:mm")}")
```

Maybe for `.kte` templates a different syntax for interpolation (e.g. `#{}` like in Ruby's ERB) would be nice to avoid confusion with Kotlin's interpolation syntax.

**Comment by casid** [comment]
Hi @edrd-f! Thank you for the heads up and for using jte.

I just pushed a bugfix so that the parser isn't confused by nested strings. This should fix the compilation issue you experienced.

The IntelliJ plugin already handles this correctly when it parses the AST:
<img src="https://user-images.githubusercontent.com/4077354/130344660-1a11bbb9-db90-41bc-a19c-34ee89d4987a.png" width="400" height="210">

I think the difference between the two is significant for the reader when using the plugin.

 jte tries to keep the syntax as minimal as possible, so I'd rather avoid adding alternative syntax for `${}`.

How urgent is this fix? I could certainly release a bugfix release to maven central today.

**Comment by edrd-f** [comment]
Wow, that was quick!

The new syntax highlighting on the plugin will help a lot too. 

About your question on urgency, it's not urgent at all, since there's the `!{}` workaround. 

Thanks a lot for your assistance.

**Comment by casid** [comment]
You're very welcome :-)

I just published version 1.11.3 to maven central, which contains this fix.

Thank you for reporting the issue!
